sstable/src/dictionary.rs

```diff
@@ -524,6 +637,222 @@ mod tests {
+
+            builder.insert(b"bbb", &1).unwrap();
+            builder.insert(b"ddd", &2).unwrap();
+
+            let table = builder.finish().unwrap();
+            let table = Arc::new(PermissionedHandle::new(table));
+            let slice = common::file_slice::FileSlice::new(table.clone());
+
+            Dictionary::<MonotonicU64SSTable>::open(slice).unwrap()
+        };
+
+        assert_eq!(dict.term_ord_or_next(b"aaa").unwrap(), TermOrdHit::Next(0));
+        assert_eq!(dict.term_ord_or_next(b"bbb").unwrap(), TermOrdHit::Exact(0));
+        assert_eq!(dict.term_ord_or_next(b"bb").unwrap(), TermOrdHit::Next(0));
+        assert_eq!(dict.term_ord_or_next(b"bbbb").unwrap(), TermOrdHit::Next(1));
+        assert_eq!(dict.term_ord_or_next(b"dd").unwrap(), TermOrdHit::Next(1));
+        assert_eq!(dict.term_ord_or_next(b"ddd").unwrap(), TermOrdHit::Exact(1));
+        assert_eq!(dict.term_ord_or_next(b"dddd").unwrap(), TermOrdHit::Next(2));
+
+        // Shouldn't this be u64::MAX?
+        assert_eq!(
+            dict.term_ord_or_next(b"zzzzzzz").unwrap(),
+            TermOrdHit::Next(2)
+        );
+    }
+    #[test]
+    fn test_term_to_ord_or_next_2() {
+        let dict = {
+            let mut builder = Dictionary::<MonotonicU64SSTable>::builder(Vec::new()).unwrap();
+
+            let mut term_ord = 0;
+            builder.insert(b"bbb", &term_ord).unwrap();
+
+            // Fill blocks in between
+            for elem in 0..50_000 {
+                term_ord += 1;
+                let key = format!("ccccc{elem:05X}").into_bytes();
+                builder.insert(&key, &term_ord).unwrap();
+            }
+
+            term_ord += 1;
+            builder.insert(b"eee", &term_ord).unwrap();
+
+            let table = builder.finish().unwrap();
+            let table = Arc::new(PermissionedHandle::new(table));
+            let slice = common::file_slice::FileSlice::new(table.clone());
+
+            Dictionary::<MonotonicU64SSTable>::open(slice).unwrap()
+        };
+
+        assert_eq!(dict.term_ord(b"bbb").unwrap(), Some(0));
+        assert_eq!(dict.term_ord_or_next(b"bbb").unwrap(), TermOrdHit::Exact(0));
+        assert_eq!(dict.term_ord_or_next(b"aaa").unwrap(), TermOrdHit::Next(0));
+        assert_eq!(dict.term_ord_or_next(b"bb").unwrap(), TermOrdHit::Next(0));
+        assert_eq!(dict.term_ord_or_next(b"bbbb").unwrap(), TermOrdHit::Next(1));
+        assert_eq!(
+            dict.term_ord_or_next(b"ee").unwrap(),
+            TermOrdHit::Next(50001)
+        );
+        assert_eq!(
+            dict.term_ord_or_next(b"eee").unwrap(),
+            TermOrdHit::Exact(50001)
+        );
+        assert_eq!(
+            dict.term_ord_or_next(b"eeee").unwrap(),
+            TermOrdHit::Next(u64::MAX)
+        );
+
+        assert_eq!(
+            dict.term_ord_or_next(b"zzzzzzz").unwrap(),
+            TermOrdHit::Next(u64::MAX)
+        );
+    }
+
+    #[test]
+    fn test_term_bounds_to_ord() {
```

Comment:
this doesn't test the cases where the key is present and lower bound (both Included and Excluded)